### PR TITLE
Adds pmd rule UseEqualsToCompareStrings

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Adds rule UseEqualsToCompareStrings to pmd-ruleset and fixes raised issues

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/logic/FormController.java [hunk 1]
```diff
@@ -943,7 +943,7 @@ public String getQuestionPromptRequiredText(FormIndex index) {
                     EvaluationContext ec = new EvaluationContext(form.getEvaluationContext(),
                             treeElement.getRef());
                     Object value = xpathRequiredMsg.eval(form.getMainInstance(), ec);
-                    if (value != "") {
+                    if (value.equals("")) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/logic/FormController.java]: This is missing a `!`, right?

**Review by lognaturel — CHANGES_REQUESTED** [review]
:/ This is why these changes make me nervous and why it's good to do them one at a time so they're easier to review.

File: collect_app/src/main/java/org/odk/collect/android/logic/FormController.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/logic/FormController.java]: fixed

**Comment by shobhitagarwal1612** [comment]
duplicate #1465

**Comment by mmarciniak90** [comment]
@opendatakit-bot unlabel "needs testing"

**Pull request closed without merging**